libs/ui-lib-tests/cypress/views/customManifestsPage.ts

```diff
@@ -0,0 +1,35 @@
+export const customManifestsPage = {
+  getFileName: () => {
+    return cy.findByTestId('filename-0');
+  },
+  getFileNameError: () => {
+    return cy.get('div.pf-c-form__helper-text.pf-m-error');
+  },
+  getStartFromScratch: () => {
+    return cy.get('.pf-c-file-upload .pf-c-empty-state__secondary > button');
+  },
+  fileUpload: () => {
+    return cy.get('input[type="file"]');
+  },
```

Comment:
`fileUpload` and `lastFileUpload` again do the same thing. It would be great if we could have one function with and index parameter that can be reused. (This is where the hierarchy-based class approach would have been useful.)
Add a test-id to the parent component (`ExpandedManifest` probably ) of the upload field if you need to.

Reply:
It would be better to have just one function for selecting the file upload inside a specific manifest.

Reply:
I've tried to add data-testid but not works.

Reply:
I don't mean a test-id. I mean doing something like `cy.get('#manifest-${index} input[type="file"]')` instead of the CSS `:last` selector. For better control over which manifest we work with.